pop-api/src/v0/fungibles/mod.rs

```diff
@@ -0,0 +1,352 @@
+//! The `fungibles` module provides an API for interacting and managing fungible tokens.
+//!
+//! The API includes the following interfaces:
+//! 1. PSP-22
+//! 2. PSP-22 Metadata
+//! 3. Management
+//! 4. PSP-22 Mintable & Burnable
+pub mod errors;
+pub mod events;
+pub mod interfaces;
+#[cfg(test)]
+mod tests;
+
+use crate::{
+	constants::{ASSETS, BALANCES, FUNGIBLES},
+	primitives::{AccountId, Balance, TokenId},
+	ChainExtensionMethodApi, StatusCode,
+};
+use constants::*;
+use core::result::Result;
+pub use errors::*;
+pub use events::*;
+use ink::prelude::vec::Vec;
+pub use interfaces::*;
+pub use management::*;
+pub use metadata::*;
+
+pub type PSP22Result<T> = Result<T, PSP22Error>;
+pub type PopApiResult<T> = Result<T, StatusCode>;
+
+// Helper method to build a dispatch call.
+//
+// Parameters:
+// - 'dispatchable': The index of the dispatchable function within the module.
+fn build_dispatch(dispatchable: u8) -> ChainExtensionMethodApi {
+	crate::v0::build_dispatch(FUNGIBLES, dispatchable)
+}
+
+// Helper method to build a call to read state.
+//
+// Parameters:
+// - 'state_query': The index of the runtime state query.
+fn build_read_state(state_query: u8) -> ChainExtensionMethodApi {
+	crate::v0::build_read_state(FUNGIBLES, state_query)
+}
```

Comment:
Can we move those + the constants module to the bottom so that readers of the code first see the apis which are the most important.